hugr-core/src/hugr/internal.rs

```diff
@@ -260,6 +281,46 @@ impl<T: RootTagged<RootHandle = Node> + AsMut<Hugr>> HugrMutInternals for T {
             .set_num_ports(node.pg_index(), incoming, outgoing, |_, _| {})
     }
 
+    fn insert_ports(
+        &mut self,
+        node: Node,
+        direction: Direction,
+        index: usize,
+        amount: usize,
+    ) -> Range<usize> {
+        let old_num_ports = match direction {
+            Direction::Incoming => self.base_hugr().graph.num_inputs(node.pg_index()),
+            Direction::Outgoing => self.base_hugr().graph.num_outputs(node.pg_index()),
+        };
```

Comment:
[`num_ports`](https://docs.rs/portgraph/latest/portgraph/view/trait.PortView.html#tymethod.num_ports)

```suggestion
        let old_num_ports = self.base_hugr().graph.num_ports(node.pg_index(), direction);
```